example/fixtures.py

```diff
@@ -53,51 +49,42 @@ def for_fixture(
 
         return cls(name, DDB_IMAGE, wait_lines, wait_timeout, kwargs)
 
+    def pytest_fixture(self, name):
+        return super().pytest_fixture(
+            name, dependencies=('db_container', 'amqp_container'))
+
     @classmethod
     def make_fixture(cls, fixture_name, name, *args, **kw):
-        return cls.for_fixture(name, *args, **kw).pytest_fixture(
-            fixture_name, dependencies=('db_container', 'amqp_container'))
+        return cls.for_fixture(name, *args, **kw).pytest_fixture(fixture_name)
 
 
-single_container = DjangoBootstrapContainer.make_fixture(
-    'single_container', 'web',
-    [r'Booting worker', r'celery@\w+ ready', r'beat: Starting\.\.\.'],
+single_container = DjangoBootstrapContainer.for_fixture(
+    'web', [r'Booting worker', r'celery@\w+ ready', r'beat: Starting\.\.\.'],
     env_extra={'CELERY_WORKER': '1', 'CELERY_BEAT': '1'})
 
 
-web_only_container = DjangoBootstrapContainer.make_fixture(
-    'web_only_container',  'web', [r'Booting worker'])
+web_container = DjangoBootstrapContainer.for_fixture(
+    'web', [r'Booting worker'])
 
 
-worker_only_container = DjangoBootstrapContainer.make_fixture(
-    'worker_only_container', 'worker', [r'celery@\w+ ready'],
+worker_container = DjangoBootstrapContainer.for_fixture(
+    'worker', [r'celery@\w+ ready'],
     command=['celery', 'worker'], publish_port=False)
 
 
-beat_only_container = DjangoBootstrapContainer.make_fixture(
-    'beat_only_container', 'beat', [r'beat: Starting\.\.\.'],
+beat_container = DjangoBootstrapContainer.for_fixture(
+    'beat', [r'beat: Starting\.\.\.'],
     command=['celery', 'beat'], publish_port=False)
 
 
-def make_multi_container(name, containers):
-    @pytest.fixture(name=name, params=containers)
+def make_multi_fixture(name, fixtures):
+    @pytest.fixture(name=name, params=fixtures)
     def containers(request):
         yield request.getfixturevalue(request.param)
     return containers
 
 
-web_container = make_multi_container(
-    'web_container', ['single_container', 'web_only_container'])
-
-worker_container = make_multi_container(
-    'worker_container', ['single_container', 'worker_only_container'])
-
-beat_container = make_multi_container(
-    'beat_container', ['single_container', 'beat_only_container'])
-
-
-__all__ = [
-    'raw_db_container', 'db_container', 'raw_amqp_container',
-    'amqp_container', 'single_container', 'web_only_container',
-    'worker_only_container', 'beat_only_container', 'web_container',
-    'worker_container', 'beat_container']
+def make_combined_fixture(base):
```

Comment:
Maybe a small comment about what's going on here or in `make_multi_fixture`?

Reply:
Will do. I'll probably just combine the two functions.